### PR TITLE
[SIG-4705] changed svg to styled component button

### PR DESCRIPTION
**Pull request by vdegraaf**
Ticket: [SIG-4705](https://datapunt.atlassian.net/browse/SIG-4705)

## Context
The error message of the My Location could not be accessed with a keyboard.

## Changes
Changed it so it makes use of the styled component Button

## Open Points
- Test fails because it doesn't recognise the variant="primary". Why is that?
- There are several ways buttons are implemented in the code. What is the convention at for this application?

### CODE REVIEW

**Comment by kiruys** [comment]
The test fails because it is not possible to use the primary variant together with the icon attribute. if you want to use the primary variant, you can use iconLeft to add the icon.

**Comment by kiruys** [comment]
The preference is to use the amsterdam styled components buttons

File: src/signals/incident/components/form/MapSelectors/components/MapMessage/MapMessage.tsx [hunk 1]
```diff
@@ -82,7 +86,7 @@ export const ZoomMessage: FC<PropsWithChildren<ZoomMessageProps>> = ({
 
 interface MapMessageProps
   extends PropsWithChildren<Omit<HTMLAttributes<HTMLDivElement>, 'onClick'>> {
-  onClick: MouseEventHandler<SVGElement>
+  onClick: () => void
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/components/MapMessage/MapMessage.tsx:89]: kan nog beschrijvender met MouseEventHandler<HTMLElement>

File: src/signals/incident/components/form/MapSelectors/components/MapMessage/MapMessage.tsx [hunk 2]
```diff
@@ -92,6 +96,6 @@ export const MapMessage: FC<MapMessageProps> = ({
 }) => (
   <MessageOverlay {...props} data-testid="mapMessage" type="map">
     {children}
-    <CloseIcon onClick={onClick} width={22} fill="white" role="button" />
+    <CloseButton onClick={onClick} icon={<Close />} variant='primary' aria-label='Melding sluiten' />
```
Comment (OscarBakker) [inline src/signals/incident/components/form/MapSelectors/components/MapMessage/MapMessage.tsx:99]: in html is het eigenlijk gebruikelijk om double quotes te gebruiken

File: src/signals/incident/components/form/MapSelectors/components/MapMessage/MapMessage.tsx (hunk 1 above)
Comment (vdegraaf) [reply src/signals/incident/components/form/MapSelectors/components/MapMessage/MapMessage.tsx]: Heb dit nu als type: MouseEventHandler<HTMLAnchorElement> & MouseEventHandler<HTMLButtonElement>
Hoewel het descriptive is, is het ook veel meer code / onleesbaarder. Op andere plekken in de code wordt vaker () => void gebruikt voor onClick.

**Review by kiruys — APPROVED** [review]
Ziet er goed uit!

oh noes, should be:  Looks good to me!

**Pull request merged**